active/PagerControl/PagerControl.md

```diff
@@ -0,0 +1,336 @@
+
+### Design Components and Functionality
+
+ | Component |  Notes |
+|:---:|:---|
+| LastButton | * Button displaying text and/or glyph indicating that the user may navigate to the last index. <br> * Automatically disabled when at last index. <br> * Can be set to not be visible when at the last index.  <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760392-3a0d6980-fc79-11e8-8d96-a811cee1c759.png) |
+| FirstButton | * Button displaying text and/or glyph indicating that the user may navigate to the first index. <br> * Automatically disabled when at first index. <br> * Can be set to not be visible when at the first index.  <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760409-485b8580-fc79-11e8-9545-a580f6016f6c.png) |
+| NextButton | * Button displaying text and/or glyph indicating that the user may navigate to the next index. <br> * Automatically disabled when at last index. <br> * Can be set to not be visible when at the last index. <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760426-54474780-fc79-11e8-9446-ee2886941a24.png) |
+| PreviousButton | * Button displaying text and/or glyph indicating that the user may navigate to the previous index. <br> * Automatically disabled when at first index. <br> * Can be set to not be visible when at the first index.  <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760443-62956380-fc79-11e8-8f93-562a92fde666.png) |
+| Ellipsis | * Button, reading "...", used before or after the first/last index to indicate an accessible but omitted range of indexes. <br> * Visibility of the first/last index can be disabled. <br> <br> * Only visible when using button panel as the display mode.  <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760513-85277c80-fc79-11e8-926e-3453ea29b0a3.png) |
+| PrefixText | * Text displayed before the ComboBox or NumberBox indexing component. <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760540-93759880-fc79-11e8-9c4f-3c3222b95a13.png) |
+| SuffixText | * When a total number of indices (N) is given, this suffix string will appear after the ComboBox indexing component or NumberBox component and read "of N". Localization will put "N" where it should be in a given language. <br><br> ![image](https://user-images.githubusercontent.com/16964652/49760557-9a9ca680-fc79-11e8-943f-3425d993fdeb.png) |
+
+## API Details
+
+```c++ 
+enum PagerDisplayMode
+{
+    Auto,
+    ComboBox,
+    NumberBox,
+    ButtonPanel,
+};
+
+enum PagerButtonVisibilityBehavior
+{
+    Auto,
+    AlwaysVisible,
+    HiddenOnEdge,
+    None,
```

Comment:
I am not sure if I think Auto is the best description for what it actually does. In a lot of cases, Auto actually has some non static behavior, e.g. [NavigationViewPaneDisplayMode.Auto](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.navigationviewpanedisplaymode?view=winui-2.4). Maybe `Default` is a better name?

Reply:
Without having read to deep into this, wouldn't this be a case similar to the [TabViewCloseButtonOverlayMode](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.tabviewclosebuttonoverlaymode?view=winui-2.4) API? If so, since we used used `Auto` there and not `Default`, we should also use `Auto` here.

Reply:
It would be the same case as the TabViewCloseButtonOverlayMode, however we also have API's that use Auto in a different way, which makes it confusing for developers to consume APIs given that Auto might be the system default that maps to another value or it might be something different.

Reply:
I think we should stay consistent here within the current API system - and apparently there weren't any real concerns to use `Auto` for the `TabViewCloseButtonOverlayMode` API preventing this name to be used - and I remember `Auto` was discussed a bit. So I favor going with `Auto` here, instead of introducing a new naming scheme, introducing even more inconsistency.

Reply:
That's true. Maybe we should stick with Auto then.

Reply:
I have considered changing it to default since there is no logic behind what Auto would do in this case but I agree that there should be some sort of consistency across the controls. @StephenLPeters do you have a recommendation based on the other controls?

Reply:
After talking internally we decided to keep it as Auto since that is what the majority of the other controls use. I know it's not consistent behavior for every control but in this case we are choosing Auto as the consistent term.

Reply:
The guideline is to use 'Auto' for cases where the control decides for you, which is what this does.  It's not required to by dynamic. (Isn't this dynamic, though, if you change NumberOfPages?)

We've violated that guideline a bunch in the past, but we're trying to stick to it on new APIs.